### PR TITLE
lvgl: kscan: set touch points within range

### PR DESCRIPTION
**Pull request by papadkostas**
Check whether input points are outside of display and set to the maximum of the its axis.
Currently when an input device sends a point out of range, lvgl reports that it's outside of it.
I've implemented basic support for stmpe811 but the display readings are not 100% accurate.
So on the edges the point reported exceeds most of the times the actual display width and height.
That's a workaround avoiding implementation of input device calibration.
Maybe a calibration functionality could be implemented within kscan, rather than doing for each different input device driver?

### CODE REVIEW

**Comment by papadkostas** [comment]
@brgl @kisvegabor

**Comment by brgl** [comment]
This looks good. I sometimes see these myself too.

**Review by brgl — APPROVED** [review]

File: zephyr/lvgl.c [hunk 1]
```diff
@@ -271,6 +271,19 @@ static void lvgl_pointer_kscan_read(lv_indev_drv_t *drv, lv_indev_data_t *data)
...
+	if(prev.point.x <= 0) {
+		prev.point.x = 0;
+	} else if (prev.point.x >= cap.x_resolution) {
+		prev.point.x = cap.x_resolution-1;
+	}
+
+	if(prev.point.y<= 0) {
+		prev.point.y = 0;
+	} else if (prev.point.y >= cap.y_resolution) {
+			prev.point.y = cap.y_resolution-1;
```
Comment (kisvegabor) [inline zephyr/lvgl.c:284]:
```suggestion
		prev.point.y = cap.y_resolution-1;
```

**Review by kisvegabor — COMMENTED** [review]
Looks reasonable!

**Comment by papadkostas** [comment]
> This looks good. I sometimes see these myself too.

You're probably also using resistive panel? My F769 discovery doesn't give me any out of range coordinates but it's capacitive FT6202. This PR is after I tried lvgl on F429 with stmpe811, will soon add also support for XPT2046 to see how that's going to perform.

> Looks reasonable!

Yes, for the time is okay(for resistive panels), but I think it would be better integrating a calibration functions within either lvgl or kscan on zephyr. Althrough, I think that when resistive panels will be extinguished, that will be useless. Maybe would be better to add a symbol to enable it for specific models? like 
`
#ifdef CONFIG_KSCAN_STMPE811 || CONFIG_KSCAN_XPT2046
`

**Review by brgl — CHANGES_REQUESTED** [review]

File: zephyr/lvgl.c [hunk 2]
```diff
@@ -271,6 +271,19 @@ static void lvgl_pointer_kscan_read(lv_indev_drv_t *drv, lv_indev_data_t *data)
 		prev.point.y = x;
 	}
 
+	/* filter readings within display */
+	if(prev.point.x <= 0) {
```
[the remark this answers is not in the document]
Comment (papadkostas) [reply zephyr/lvgl.c]: Done.

File: zephyr/lvgl.c [hunk 3]
```diff
@@ -271,6 +271,19 @@ static void lvgl_pointer_kscan_read(lv_indev_drv_t *drv, lv_indev_data_t *data)
...
 	}
 
+	/* filter readings within display */
+	if (prev.point.x <= 0) {
+		prev.point.x = 0;
+	} else if (prev.point.x >= cap.x_resolution) {
+		prev.point.x = cap.x_resolution-1;
+	}
+
+	if (prev.point.y<= 0) {
```
Comment (brgl) [inline zephyr/lvgl.c:281]: Still one missing space here

**Review by brgl — CHANGES_REQUESTED** [review]

**Review by brgl — APPROVED** [review]

**Comment by kisvegabor** [comment]
FYI, we use [this script](https://github.com/zephyrproject-rtos/lvgl/blob/zephyr/scripts/code-format.py) to format code.

**Comment by brgl** [comment]
> > FYI, we use [this script](https://github.com/zephyrproject-rtos/lvgl/blob/zephyr/scripts/code-format.py) to format code.
> 
> @kisvegabor Thanks for the info. I tried the script and it removes the space after an if statement.

Wait wat?? No spaces after `if`? I'm all for keeping the code formatting of the original project, but come on, this is worse than the GNU coding style. :)

**Comment by kisvegabor** [comment]
> do you have a CI pipeline for code formatting in your main repo? Maybe we should port it here.

No, because we couldn't come up with good idea how to handle format errors:
- push to the authors repo? not nice to push to someones repo automatically.  
- create a patch as an artifact that someone can apply? no one found and used that patch
- add a comment with the diff of the formatting? still not that useful. 

So we kept it just to let the CI fail until we can come up with a good idea. (So ideas are welcome :slightly_smiling_face:  )


> Wait wat?? No spaces after if? I'm all for keeping the code formatting of the original project, but come on, this is worse than the GNU coding style. :)

Really?  I do it with no space since I'm programming :laughing: I believe it's because the default K&R style of Eclipse adds no space.  
You really [have space](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/crypto/crypto_nrf_ecb.c#L45) after `if`.  And [here](https://en.wikipedia.org/wiki/Indentation_style) all styles have have space. Considering... :slightly_smiling_face: 

(cc @embeddedt )

**Comment by brgl** [comment]
I've merged it, coding style can be decided on later. I'll open a separate issue.

**Comment by kisvegabor** [comment]
@papadkostas 
Strange, `os.system('astyle --options=code-format.cfg "../zephyr/*.c,*.h"')` seems correct. :confused:

**Pull request merged**